DataFormats/Scouting/test/TestReadRun3Scouting.cc

```diff
@@ -0,0 +1,1091 @@
+// -*- C++ -*-
+//
+// Package:    DataFormats/Scouting
+// Class:      TestReadRun3Scouting
+//
+/**\class edmtest::TestReadRun3Scouting
+  Description: Used as part of tests that ensure the run 3 Scouting
+  data formats can be persistently written and in a subsequent process
+  read. First, this is done using the current release version for writing
+  and reading. In addition, the output file of the write process should
+  be saved permanently each time a run 3 Scouting persistent data
+  format changes. In unit tests, we read each of those saved files to verify
+  that the current releases can read older versions of these data formats.
+*/
+// Original Author:  W. David Dagenhart
+//         Created:  18 May 2023
+
+#include "DataFormats/Scouting/interface/Run3ScoutingCaloJet.h"
+#include "DataFormats/Scouting/interface/Run3ScoutingElectron.h"
+#include "DataFormats/Scouting/interface/Run3ScoutingHitPatternPOD.h"
+#include "DataFormats/Scouting/interface/Run3ScoutingMuon.h"
+#include "DataFormats/Scouting/interface/Run3ScoutingParticle.h"
+#include "DataFormats/Scouting/interface/Run3ScoutingPFJet.h"
+#include "DataFormats/Scouting/interface/Run3ScoutingPhoton.h"
+#include "DataFormats/Scouting/interface/Run3ScoutingTrack.h"
+#include "DataFormats/Scouting/interface/Run3ScoutingVertex.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDAnalyzer.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include <vector>
+
+namespace edmtest {
+
+  class TestReadRun3Scouting : public edm::global::EDAnalyzer<> {
+  public:
+    TestReadRun3Scouting(edm::ParameterSet const&);
+    void analyze(edm::StreamID, edm::Event const&, edm::EventSetup const&) const override;
+    static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+  private:
+    void analyzeCaloJets(edm::Event const&) const;
+    void analyzeElectrons(edm::Event const&) const;
+    void analyzeMuons(edm::Event const&) const;
+    void analyzeParticles(edm::Event const&) const;
+    void analyzePFJets(edm::Event const&) const;
+    void analyzePhotons(edm::Event const&) const;
+    void analyzeTracks(edm::Event const&) const;
+    void analyzeVertexes(edm::Event const&) const;
+
+    void throwWithMessage(const char*) const;
+
+    // These expected values are meaningless other than we use them
+    // to check that values read from persistent storage match the values
+    // we know were written.
+
+    std::vector<double> expectedCaloJetsValues_;
```

Comment:
Seems like all of these could be `const`

Reply:
I made all the data members const in both modules.